### ISSUE TITLE
EdfWriter writes incorrect sample frequencies for record lengths != 1

### ISSUE THREAD

**Issue by cfranklin11** [holgern/pyedflib#117]
I'm working with EDF files that have record lengths other than 1 second and am trying to generate dummy files for testing, and I think I've encountered a bug with how `EdfWriter` handles sample rate vs sample frequency. I'm writing a file with a sample rate of 256, a record length of 2, and a total of 6 minutes of data (example code below). I expect the generated EDF file to have 180 records with 512 samples per record, but the actual file has 360 records with 256 samples per record.

I think I found the source of the problem with regards to calculating sample frequency: in `EdfWriter.update_header`, channel sample rates are passed to `set_samplefrequency`, but inside that function the argument is named `samplefrequency` and treated as such. My understanding is that sample rate represents samples-per-second, and sample frequency represents samples-per-record. So treating them interchangeably would result in incorrect numbers for record lengths != 1. I saw #111 and suspect that this is related.

I haven't found where number of records is calculated yet, but I imagine the issue is similar for that header attribute as well.

Example code:
```
import numpy as np
from pyedflib import EdfWriter, FILETYPE_EDF
from datetime import datetime

MILLISECONDS = 1000
TEN_MICROSECONDS = MILLISECONDS * 100
SIX_MINUTES = 360

filename = 'edf_file.edf'
sample_rate = 256
samples_per_record = 512  # For a 2 sec record length
record_length = (samples_per_record / sample_rate) * TEN_MICROSECONDS
channel_names = ['ch1', 'ch2', 'ch3', 'ch4', 'ch5']
signals = np.random.rand(len(channel_names), sample_rate * SIX_MINUTES) * 200
start_date = datetime(2017, 5, 24, 15, 53, 49)

writer = EdfWriter(filename, len(channel_names), file_type=FILETYPE_EDF)
writer.setPatientCode('')
writer.setRecordingAdditional('')
writer.setDatarecordDuration(record_length)
writer.setStartdatetime(start_date)
writer.setSignalHeaders([{
    'label': channel,
    'sample_rate': sample_rate,
    'dimension': 'uV',
    'physical_min': -200.0,
    'physical_max': 200.0,
    'digital_min': -32768,
    'digital_max': 32767,
    'transducer': f'EEG {channel}',
    'prefilter': ''
} for channel in channel_names])

writer.writeSamples(signals)
```

Let me know if my assessment is correct (I'm new to working with EDF files, so I'm still getting across how they're structured), and I'd be happy to open a PR.

**Comment by skjerns** [comment]
Excuse my late reply, I did not have time to look into this in more detail yet, and fear that I will not have time soon as well.

However, I think you are on a good track! It seems very much related to the issue I raised in #111 . I did not implement the piece of code mentioned above, but as far as I understand it, the sample frequency is never saved explicitly in the header, but always computed by `duration of a data record` (file header) and `nr of samples in each data record` (signal header). Therefore it can become a bit confusing when specifying a sampling frequency that is transformed into the former internally.

I'd very much appreciate if you have a look into this and post your results and a PR

Thanks alot!

**Comment by cfranklin11** [comment]
No worries. I was able to isolate the two places in the code where record length is calculated or used in other calculations, and implemented a quick fix in a version of `pyedflib` installed from local files. I've since used it to write and read EDF files with irregular record lengths, and the data and headers all turn out correct.

I'll look into cleaning up my fixes and adding tests for a PR sometime in the near future.
